test/SendRecv_SinglePairs.cpp

```diff
@@ -0,0 +1,86 @@
+/*************************************************************************
+ * Copyright (c) 2022 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * See LICENSE.txt for license information
+ ************************************************************************/
+#include "TestBed.hpp"
+
+namespace RcclUnitTesting
+{
+  TEST(SendRecv, SinglePairs)
+  {
+    TestBed testBed;
+
+    // Configuration
+    std::vector<ncclDataType_t> const& dataTypes       = {ncclInt32, ncclFloat64};
+    std::vector<int>            const  numElements     = {1048576, 53327, 1024};
+    bool                        const  inPlace         = false;
+    bool                        const  useManagedMem   = false;
+
+    int numCollPerGroup = 0;
+    bool isCorrect = true;
+    int totalRanks = testBed.ev.maxGpus;
+    for (int isMultiProcess = 0; isMultiProcess <= 1 && isCorrect; ++isMultiProcess) // CHANGE TO SINGLE
+    {
+      int const numProcesses = isMultiProcess ? totalRanks : 1;
+      testBed.InitComms(TestBed::GetDeviceIdsList(numProcesses, totalRanks), 1);
+
+      for (int dataIdx = 0; dataIdx < dataTypes.size() && isCorrect; ++dataIdx)
+      {
+        for (int numIdx = 0; numIdx < numElements.size() && isCorrect; ++numIdx)
+        {
+          // Run all element sizes in parallel as single group
+          for (int sendRank = 0; sendRank < totalRanks; ++sendRank)
+          {
+            for (int recvRank = 0; recvRank  < totalRanks; ++recvRank)

```

Comment:
Collapse these for loops down if possible to cut down on indentation levels / bracketing.  See https://github.com/ROCmSoftwarePlatform/rccl/blob/develop/test/common/TestBed.cpp#L417